iroh-net/src/magicsock.rs

```diff
@@ -3544,4 +3608,276 @@ mod tests {
+        let node_id_missing_node = secret_key_missing_node.public();
+
+        let (ep_1, msock_1) = magicsock_ep(secret_key_1.clone()).await.unwrap();
+
+        // Generate an address not present in the NodeMap.
+        let bad_addr = QuicMappedAddr::generate();
+
+        // 500ms is rather fast here.  Running this locally it should always be the correct
+        // timeout.  If this is too slow however the test will not become flaky as we are
+        // expecting the timeout, we might just get the timeout for the wrong reason.  But
+        // this speeds up the test.
+        let res = tokio::time::timeout(
+            Duration::from_millis(500),
+            magicsock_connect(&ep_1, secret_key_1.clone(), bad_addr, node_id_missing_node),
+        )
+        .await;
+        assert!(res.is_err(), "expecting timeout");
+
+        // Now check we can still create another connection with this endpoint.
+        let (ep_2, msock_2) = magicsock_ep(secret_key_2.clone()).await.unwrap();
+
+        // This needs an accept task
+        let accept_task = tokio::spawn({
+            async fn accept(ep: quinn::Endpoint) -> Result<()> {
+                let incoming = ep.accept().await.ok_or(anyhow!("no incoming"))?;
+                let _conn = incoming.accept()?.await?;
+
+                // Keep this connection alive for a while
+                tokio::time::sleep(Duration::from_secs(10)).await;
+                info!("accept finished");
+                Ok(())
+            }
+            let ep_2 = ep_2.clone();
+            async move {
+                if let Err(err) = accept(ep_2).await {
+                    error!("{err:#}");
+                }
+            }
+            .instrument(info_span!("ep2.accept, me = node_id_2.fmt_short()"))
+        });
+        let _accept_task = AbortingJoinHandle::from(accept_task);
+
+        let node_addr_2 = NodeAddr {
+            node_id: node_id_2,
+            info: AddrInfo {
+                relay_url: None,
+                direct_addresses: msock_2
+                    .direct_addresses()
+                    .next()
+                    .await
+                    .expect("no direct addrs")
+                    .into_iter()
+                    .map(|x| x.addr)
+                    .collect(),
+            },
+        };
+        msock_1
+            .add_node_addr(node_addr_2, Source::NamedApp { name: "test" })
+            .unwrap();
+        let addr = msock_1.get_mapping_addr(node_id_2).unwrap();
+        let res = tokio::time::timeout(
+            Duration::from_secs(10),
+            magicsock_connect(&ep_1, secret_key_1.clone(), addr, node_id_2),
+        )
+        .await
+        .expect("timeout while connecting");
+
+        // aka assert!(res.is_ok()) but with nicer error reporting.
+        res.unwrap();
+
+        // TODO: Now check if we can connect to a repaired ep_3, but we can't modify that
+        // much internal state for now.
```

Comment:
Wdym "repaired `ep_3`"?
Isn't `ep_2` non-broken?
I guess when the bug existed, this would've made sense, but now that the test is fixed, maybe we can just remove this TODO?

Reply:
oh, this was meant to repair what is behind `bad_addr`.  But that doesn't even have a NodeId right now.   Yeah, let's remove this.